### DISCUSSION TITLE
Where can I find detekt samples?

### DISCUSSION THREAD

**Discussion by eddynaka** [detekt/detekt#3440] [General]
Hi,

we are creating a converter in https://github.com/microsoft/sarif-sdk to generate a sarif file from a detekt file.

I searched but I couldn't find examples to test.

Can you generate a few, pls?

Thank you!

cc: @yongyan-gh

**Comment by BraisGabin** [comment]
Can you elaborate? Samples of what? Open source projects that uses detekt? A sarif report generated by detekt? Other type of report?

**Reply by eddynaka** [reply]
Hi @BraisGabin , some detekt samples. So, imagine that we run detekt in this repository, it generates a detekt file. So i want that, if you can share.

**Comment by cortinico** [comment]
> we are creating a converter in [microsoft/sarif-sdk](https://github.com/microsoft/sarif-sdk) to generate a sarif file from a detekt file.

Just to clarify, Detekt already supports exports to sarif: https://github.com/detekt/detekt/issues/3045

To answer your original questions. You can use the detekt repo itself, where we apply the tool to ourself. Otherwise, a clean setup is available here https://github.com/cortinico/kotlin-android-template

**Reply by eddynaka** [reply]
Hi @cortinico , i know :) i pushed some changes fixing some issues.
The idea is to facilitate in case the user is familiar with one tool that is called sarif.multitool, which can do some conversions.

**Reply by eddynaka** [reply]
do you have the file itself somewhere? I don't have kt here and I'm not very familiar with that.

**Reply by cortinico** [reply]
Ok so you're asking for the output in the Sarif format. That was not clear from your original message. 

The output looks like this:
```json
{
  "$schema" : "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
  "version" : "2.1.0",
  "runs" : [ {
    "tool" : {
      "driver" : {
        "guid" : "022ca8c2-f6a2-4c95-b107-bb72c43263f3",
        "name" : "detekt",
        "organization" : "detekt",
        "fullName" : "detekt",
        "version" : "1.15.0",
        "semanticVersion" : "1.15.0",
        "downloadUri" : "https://github.com/detekt/detekt/releases/download/v1.15.0/detekt",
        "informationUri" : "https://detekt.github.io/detekt",
        "rules" : [ ],
        "language" : "en"
      }
    },
    "results" : [ {
      "ruleId" : "detekt.style.MagicNumber",
      "message" : {
        "text" : "This expression contains a magic number. Consider defining it to a well named constant."
      },
      "locations" : [ {
        "physicalLocation" : {
          "artifactLocation" : {
            "uri" : "/git/kotlin-android-template/app/src/main/java/com/ncorti/kotlin/template/app/MainActivity.kt"
          },
          "region" : {
            "startLine" : 18,
            "startColumn" : 17
          }
        }
      } ]
    } ]
  } ]
}
```

**Reply by eddynaka** [reply]
do u have the detekt format as well?

**Reply by cortinico** [reply]
I'm unsure what you mean with _detekt format_, I supposes it's XML.
Supported formats are listed here https://detekt.github.io/detekt/reporting.html

Anyway the XML would look like this

```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
<file name="/git/kotlin-android-template/app/src/main/java/com/ncorti/kotlin/template/app/MainActivity.kt">
	<error line="18" column="17" severity="warning" message="This expression contains a magic number. Consider defining it to a well named constant." source="detekt.MagicNumber" />
</file>
</checkstyle>
```

**Reply by eddynaka** [reply]
interesting!
thanks for the xml file (i didn't know that detekt was exporting to XML)

**Reply by BraisGabin** [reply]
I think that we didn't understand your original message because we don't have our own file standard. We support different reports and xml is checkstyle. But we have reports in plain text and html too.
